src/components/slide_in_panel.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+import { Motion, spring } from 'react-motion';
+import * as classNames from 'classnames';
+
+import CloseButton from "./close_button";
+
+export interface SlideInPanelProps {
+  /** Text to be displayed in the panel header */
+  headerTitle: string;
+  /** The panel will be open when true */
+  isOpen: boolean;
+  /** Called when the background overlay is clicked */
+  onOverlayClickHandler?: Function;
+  /** Width of panel. Include unit */
+  panelWidth?: string;
```

Comment:
The name `width` would suffice I think as it's already in the context of the slide-in panel.